common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -480,19 +547,50 @@ func (s *VisibilityStore) GetWorkflowExecution(
 func (s *VisibilityStore) BuildSearchParametersV2(
 	request *manager.ListWorkflowExecutionsRequestV2,
 	getFieldSorter func([]elastic.Sorter) ([]elastic.Sorter, error),
+) (*client.SearchParameters, error) {
+	return s.buildSearchParametersInternal(&searchParametersInternal{
+		NamespaceName: request.Namespace,
+		NamespaceID:   request.NamespaceID,
+		Query:         request.Query,
+		PageSize:      request.PageSize,
+		NextPageToken: request.NextPageToken,
+		ChasmMapper:   nil,
+		ArchetypeID:   chasm.UnspecifiedArchetypeID,
+	})
+}
+
+func (s *VisibilityStore) BuildChasmSearchParameters(
+	request *manager.ListChasmExecutionsRequest,
+	getFieldSorter func([]elastic.Sorter) ([]elastic.Sorter, error),
+	chasmMapper *chasm.VisibilitySearchAttributesMapper,
+) (*client.SearchParameters, error) {
+	return s.buildSearchParametersInternal(&searchParametersInternal{
+		NamespaceName: request.Namespace,
+		NamespaceID:   request.NamespaceID,
+		Query:         request.Query,
+		PageSize:      request.PageSize,
+		NextPageToken: request.NextPageToken,
+		ChasmMapper:   chasmMapper,
+		ArchetypeID:   request.ArchetypeID,
+	})
+}
+
+func (s *VisibilityStore) buildSearchParametersInternal(
+	params *searchParametersInternal,
 ) (*client.SearchParameters, error) {
 	var queryParams *esQueryParams
 	var err error
 	if s.enableUnifiedQueryConverter() {
-		queryParams, err = s.convertQuery(request.Namespace, request.NamespaceID, request.Query)
+		queryParams, err = s.convertQuery(params.NamespaceName, params.NamespaceID, params.Query, params.ChasmMapper, params.ArchetypeID)
 		if err != nil {
 			return nil, err
 		}
 	} else {
 		queryParamsLegacy, err := s.convertQueryLegacy(
-			request.Namespace,
-			request.NamespaceID,
-			request.Query,
+			params.NamespaceName,
+			params.NamespaceID,
+			params.Query,
+			params.ChasmMapper,
```

Comment:
Don't you need to pass the archetype id as well for building the `TemporalNamespaceDivision` filter?